### DISCUSSION TITLE
server side query using JS lib

### DISCUSSION THREAD

**Discussion by dcsan** [supabase/supabase#3725] [Questions]
hi! I want to run some batch tasks against the DB fully server side, ie without any user / client login.

However using the JS files, I get no response from the DB

```
// lib/dbConn
require('dotenv').config()
const SUPABASE_URL: string = process.env.SUPABASE_URL!
const SUPABASE_ANON_KEY: string = process.env.SUPABASE_ANON_KEY!

import { createClient } from '@supabase/supabase-js'
const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY)
export default supabase
```

```
import supabase from '../lib/DbConn'

class Tree {

    static async test() {
        console.log('get trees')
        let { data: todos, error } = await supabase
            .from('trees')
            .select('*')
        console.log('result', { todos, error })
    }

}

export default Tree
```

calling `await Tree.test()` from elsewhere just results in `get trees...` hanging ie there is never any response to the await block (no error either)

should I be auth'ing with an ANON key or some other method?

are there any references for supabase server side? every examples seems to require boatloads of nextJS or other javascript client projects...

**Comment by MildTomato** [comment]
Do you have row level security enabled for the table 'trees' ?

If you have, then the anon key won't be able to read from the table unless a user is authenticated, or if you are using the service key.

Did `error` return anything in the console.log?

**Reply by dcsan** [reply]
I tried it with and without RLS. no error or anything is returned in the console, just a hanging connection and eventually a timeout

**Reply by dcsan** [reply]
I gave up on the anon key and tried it with the main "this will override RLS!!!!!" key, but still no response. I'm having so many problems getting this working for even basic stuff, the onboarding DX for server side code seems like virgin ground.

**Reply by inian** [reply]
Hey dcsan, can you send your project reference here? Is the connection timing out or are you not getting any data back?

**Comment by fergusmeiklejohn** [comment]
I'm not using NextJS, we're using Remix https://remix.run 
Remix has Loaders which run server side only and are called for every route before it hits the client. I'm querying the DB using the Supabase JS library with no problem, calling views like eg:
```ts
  const {data, error} = await supabase
    .from("fetch_data")
    .select()
    .eq("id", id);
```
We've not set RLS yet.

**Reply by binajmen** [reply]
Hi @fergusmeiklejohn,

I'm planning to use Remix + Supabase, but wondering how RLS can be applied if db querying is done server side. Do you use `service_role` api key since the `loader`s and `action`s are run server side? Did you manage to use RLS?

Your feedback would be very valuable 🙏

**Reply by fergusmeiklejohn** [reply]
Hi @binajmen 
Yeah I think: handle authorisation with our own logic in the action or the loaders and query the DB with the `service role`.. but like the bigger question is whether Supabase is a good fit for Remix...? Because it's designed to work from the client, Supabase seems to me to suit SPAs and mobile apps, while the happy path with Remix is more like a Rails or PHP app. So for instance with SB how do we do authentication on the server? And we'd only use the SB JS lib if we were doing real time stuff on the client (where it would be super useful). SB has storage, but then so does AWS or CloudFlare.
I think SB is great, I'd like to use it on a mobile project I'm working on, but for Remix apps I'm just not sure anymore.. I wonder what you think?

**Reply by binajmen** [reply]
I fully agree with your analysis. Although authentification with Supabase seems well handled in the package [remix-auth-supabase](https://github.com/mitchelvanbever/remix-auth-supabase) (didn't try it yet), my main concern is I feel replacing the logic offered by RLS rules with your own logic in the actions/loaders defeat one of the purpose of Postgres/Supabase I really wanted to exploit. I lose the DX I liked in the first place.

I'm no DevOps so Supabase is also really useful for me as most (all?) the backend setup is taking over. Althouth I understand that `remix` and `architect` works well in pair. I just have no idea on the cost of a AWS db setup vs. the pricing of Supabase. I had a bad experience with Google Cloud SQL and their billing 😉 

Did you go on with Remix and a different backend stack in the end?

**Reply by binajmen** [reply]
FYI I started a discussion https://github.com/remix-run/remix/discussions/1610 about this ;)
